### PR TITLE
Android/marketplace

### PR DESCRIPTION
**Pull request by Refref18**
User can publish a goal
User can download a goal
User can search a goal according to relative and exact relationship
Users can view details of any prototypes 
Only missing issue is the download number due to ambiguity in backend side will be updated in next pr.

### CODE REVIEW

**Comment by Refref18** [comment]
Use is now able to view goals and their subgoals but not able to see the entities.

**Review by yamanogluberk — COMMENTED** [review]
I ll review the remaining part once you do the clean up

**Review by ihsangkcl — COMMENTED** [review]
I tried it. It is working except search function.

File: application/android/app/src/main/java/com/group12/beabee/models/responses/ProGoalDetail.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package com.group12.beabee.models.responses;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.io.Serializable;
+import java.util.List;
+
+public class ProGoalDetail implements Serializable {
```
[the remark this answers is not in the document]
Comment (Refref18) [reply application/android/app/src/main/java/com/group12/beabee/models/responses/ProGoalDetail.java:8]: you are right, I deleted it in the previous commit

File: application/android/app/src/main/java/com/group12/beabee/views/goals/GoalsPrototypeAdapter1.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+
+        if(viewType==0){
+            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_goal_list_marketplace, parent, false);
+            return new ViewHolder(view);
+        }
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_goal_list_marketplace, parent, false);
```
Comment (yamanogluberk) [inline application/android/app/src/main/java/com/group12/beabee/views/goals/GoalsPrototypeAdapter1.java:41]: You probably dont need the second viewType and second view holder for this adapter

**Pull request merged**